library/std/src/io/buffered/bufreader/buffer.rs

```diff
@@ -124,7 +124,6 @@ impl Buffer {
     pub fn backshift(&mut self) {
         self.buf.copy_within(self.pos.., 0);
         self.initialized -= self.pos;
-        self.filled -= self.pos;
```

Comment:
I think this is the wrong line – `initialized` should remain unchanged, but `filled` must get smaller.

Reply:
Whoops!